doc/adrs/188_add_actions_on_diagram_nodes.adoc

```diff
@@ -0,0 +1,95 @@
+= ADR-188 - Add actions on diagram nodes
+
+== Context
+
+The Show/Hide menu of the diagram nodes context menu (a.k.a the palette) is difficult to understand for the end-users.
+Furthermore, it forces users to click 3 times to hide a graphical node:
+
+- right click on the diagram node to open the palette
+- left click on the Show/Hide section
+- click on Hide menu item
+
+== Decision
+
+A new UI should be available in Sirius Web, allowing to easily execute actions on diagram nodes.
+This UI would be an icon and would be placed on the top right corner of the node.
+The specifier will tell which node descriptions will have this UI.
+
+=== Backend
+
+==== View DSL Diagram
+
+On a `diagram::NodeDescription`, it will be possible to create new `diagram::NodeAction`.
+The `diagram::NodeAction` will have four properties:
+
+- a identifier, to let the frontend highjack the rendering of the node action
+- a labelExpression, to display the label of the action as a tooltip
+- a preconditionExpression, to only make available the action on a specific condition
+- an iconURLsExpression, to display the icon of the action
+
+On a `diagram::NodeAction`, it will be possible to add the view dsl operations (e.g. change context, create instance, delete element...)
+These view dsl operations will only be executed if the identifier of the `diagram::NodeAction` can't be handled by the frontend (see Frontend section for more details).
+
+When a new `diagram::NodeDescription` will be created, a default `diagram::NodeAction` will be created at the same time (as it is already the case for the node palette).
+This default `diagram::NodeAction` will allow to automatically hide the node.
+
+==== API
+
+A new API will allow to call the view dsl operations of a `diagram::NodeAction`.
+
+.org.eclipse.sirius.components.collaborative.diagrams.api
+[code,java]
+----
+public interface INodeActionHandler {
+  boolean canHandle(IDiagramInput diagramInput);
+  void handle(IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput);
+};
+----
+
+This is approximately the same behavior than the diagram tools (i.e. `InvokeSingleClickOnDiagramElementToolEventHandler.java`)
+
+
+A new `NodeAction` class will be available:
+
+.org.eclipse.sirius.components.diagram.actions
+[code,java]
+----
+public class NodeAction {
+  public String id;
+  public String label:
+  public List<String> iconURL;
+};
+----
```

Comment:
I just thought about that and it may end up in sirius-components-collaborative-diagrams since I think that absolutely nothing will use it in sirius-components-diagrams. It's a minor details, we don't care about that in the ADR. We can merge it as is.